azurerm/data_source_arm_container_registry.go

```diff
@@ -0,0 +1,87 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmContainerRegistry() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmContainerRegistryRead,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		MigrateState:  resourceAzureRMContainerRegistryMigrateState,
+		SchemaVersion: 2,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateAzureRMContainerRegistryName,
+			},
+
+			"resource_group_name": resourceGroupNameSchema(),
+		},
```

Comment:
there's other fields being set below which don't exist in the Schema - can we add them? They'll be computed only fields since this is a Data Source, so this'll be:

```
"location": locationForDataSourceSchema(),

"admin_enabled": {
  Type: schema.TypeBool,
  Computed: true,
},

"admin_password": {
  Type: schema.TypeString,
  Computed: true,
},

"admin_username": {
  Type: schema.TypeString,
  Computed: true,
},

"login_server": {
  Type: schema.TypeString,
  Computed: true,
},

"sku": {
  Type: schema.TypeString,
  Computed: true,
},

"storage_account_id": {
  Type: schema.TypeString,
  Computed: true,
},
```